ale_linters/c/checkpatch.vim

```diff
@@ -0,0 +1,62 @@
+" Author: Matt Ihlenfield, mtihlenfield@protonmail.com
+" Description: checkpatch linter for c files
+
+let s:default_options = '--strict --no-tree'
+
+" Set the  location of the checkpatch executable. By default it assumes it
+" is in $(pwd)/scripts/checkpatch.pl
+call ale#Set('c_checkpatch_executable', '')
+
+call ale#Set('c_checkpatch_options', s:default_options)
+
+function! ale_linters#c#checkpatch#GetExecutable(buffer) abort
+    let l:checkpatch_executable = ale#Var(a:buffer, 'c_checkpatch_executable')
+
+    if l:checkpatch_executable is# '' && g:ale_use_global_executables
```

Comment:
You should write `!g:ale_use_global_executables` here. The option defaults to `0` and can be used to disable using local executables for things.